common/os_calls.c

```diff
@@ -3049,6 +3076,45 @@ g_set_allusercontext(int uid)
     return (rv != 0);  /* Return 0 or 1 */
 }
 #endif
+
+/*****************************************************************************/
+void
+g_get_executable_path(enum xrdp_exe xe, char *buf, int bufsize)
+{
+    int rv = -1;
+#if defined(__APPLE__)
+    uint32_t _bufsize = bufsize;
+#endif
+
+    g_memset(buf, '\0', bufsize);
+
+#if defined(__APPLE__)
+    rv = _NSGetExecutablePath(buf, &_bufsize);
+#elif defined(__linux__)
+    rv = readlink("/proc/self/exe", buf, bufsize);
+#endif
+
+    if (rv > 0)
+    {
+        return;
+    }
+
+    // build executable path manually
```

Comment:
You can remove everything above this line I think. You've then got common code which works on all platforms, and can be added to your unit testing. At the moment you've got 3 streams to test, and you can't cover them all on one platform.